### ISSUE TITLE
Legal review?

### ISSUE THREAD

**Issue by AllenJB** [ContributorCovenant/hippocratic-license#5]
I have serious concerns about the enforceability of this license. Who with any legal knowledge has reviewed this license to determine its enforceability? What were their full conclusions?

**Comment by Pr0methean** [comment]
https://twitter.com/lawjolla/status/1175909886376333312?s=20

"Speaking as an IP attorney, that license carries no weight. Void for vagueness."

**Comment by Daniel15** [comment]
I am not a lawyer, but I think as-is this license is unenforceable due to being overly vague and broad. Terms like "harm" and "underprivileged" should have precise definitions as part of the license text to resolve any ambiguity.

Some examples of scenarios which would need to be clarified:
- It says that the software can't be used by people that "actively and knowingly endanger, harm, or otherwise threaten". Does that mean that if someone doesn't know that their actions are harming someone, it's fine and they're not violating the license?
- Australia's power is predominantly coal-powered. It is well known that emissions from coal power plants are harmful to humans. 
  - Does the license mean that people that work at a power plant can not use the software? 
  - If yes, does that restriction only apply for their work that directly relates to their job at the power plant, or does it apply to the individuals personally too (eg. if they want to use the software in their free time)?
  - What about individuals that live in houses powered by coal power plants, since their choice to not use renewable energy sources could be indirectly affecting the well-being of people living near the power plants? How far-reaching is it?
- The license specifically mentions "underprivileged individuals". Does that mean that it'd be okay for a serial killer who only targets privileged individuals to use the software, but it wouldn't be fine if they only targeted unprivileged individuals?
- If one employee at a company actively harms underprivileged individuals, does the license revocation apply to just that one individual, or does it apply to their employer too? 

Don't get me wrong, I really do think this is a good idea, however it should be reviewed by an IP lawyer.

**Comment by quorten** [comment]
A similar but older license is the JSMin/JSLint license by Douglas Crockford.  It was also a modified MIT license with an additional ethics term.  IBM's lawyers ended up getting a waiver signed "to use JSLint for evil."

My understanding is that the enforceability of the Hippocratic License is similar.  That is, the enforceability of the classic MIT license terms will hold regardless of the ethics clause, which might be explicitly waived by some users in additional formal legal documents.

**Comment by nateberkopec** [comment]
@Daniel15 the underprivileged text was removed in 1.1.

I raised similar concerns on the Contributor Covenant here: https://github.com/ContributorCovenant/contributor_covenant/issues/284, but ultimately decided that any unwanted potential legal liability was worth the benefit of what the Contributor Covenant is trying to accomplish.

I think the case for legal review here is much more important, because a license is intended to be a legal document, and for the Hippocratic License to achieve it's purpose, it must be something that will survive legal challenge. 

Is the Hippocratic License still valuable _even if_ it is unenforceable? I'm not sure. Without the usage restriction, it's another non-free license that will inhibit the adoption of software by otherwise-good people, e.g. the various good people who are unable to use JSON because it's nonfree. There's been a lot wasted effort we've had to have creating free alternatives to JSON.

**Comment by sffc** [comment]
Are these concerns about enforceability based on the old version 1.0 or the new version 1.1 that makes the license specifically tied to the UN UDHR as the standard for ethics?

**Comment by rafaelcastrocouto** [comment]
> It says that the software can't be used by people that "actively and knowingly endanger, harm, or otherwise threaten". Does that mean that if someone doesn't know that their actions are harming someone, it's fine and they're not violating the license?

Yes. As long as they actually didn't know and stop doing as soon as they realize it.

> Australia's power is predominantly coal-powered. It is well known that emissions from coal power plants are harmful to humans.
Does the license mean that people that work at a power plant can not use the software?

No, they don't work to cause harm, they work to generate electricity. Almost every kind of human activities generate some kind of colateral damage and the coal industry does need to obey laws to minimize this harm, else they wouldn't be allowed to do it as well.

> What about individuals that live in houses powered by coal power plants, since their choice to not use renewable energy sources could be indirectly affecting the well-being of people living near the power plants? How far-reaching is it?

This makes absolutely no sense.

> The license specifically mentions "underprivileged individuals". Does that mean that it'd be okay for a serial killer who only targets privileged individuals to use the software, but it wouldn't be fine if they only targeted unprivileged individuals?

As pointed by @nateberkopec, the underprivileged text was removed in 1.1.

> If one employee at a company actively harms underprivileged individuals, does the license revocation apply to just that one individual, or does it apply to their employer too?

 If only one employee does it, then just that one individual would respond.
